From 8426569f5c6ee954c5f0345351bd8239725a8fee Mon Sep 17 00:00:00 2001
From: Martin Braun <martin.braun@ettus.com>
Date: Thu, 5 May 2016 16:32:37 -0700
Subject: [PATCH 01/29] doxygen/cmake: Allow Doxygen to use shorter filenames

This fixes cases where automatic builds fail on operating systems with
character limits in paths, such as Windows.
---
 host/docs/CMakeLists.txt | 6 ++++++
 host/docs/Doxyfile.in    | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/host/docs/CMakeLists.txt b/host/docs/CMakeLists.txt
index 96f388b..7cb0472 100644
--- a/host/docs/CMakeLists.txt
+++ b/host/docs/CMakeLists.txt
@@ -67,6 +67,7 @@ MESSAGE(STATUS "")
 LIBUHD_REGISTER_COMPONENT("API/Doxygen" ENABLE_DOXYGEN ON "DOXYGEN_FOUND" OFF OFF)
 OPTION(ENABLE_DOXYGEN_FULL "Use Doxygen to document the entire source tree (not just API)" OFF)
 OPTION(ENABLE_DOXYGEN_DOT "Let Doxygen use dot (requires graphviz)" OFF)
+OPTION(ENABLE_DOXYGEN_SHORTNAMES "Let Doxygen use shorter filenames (helps on Windows)" OFF)
 
 IF(LIBUHDDEV_PKG)
     SET(PKG_DOC_DIR share/doc/libuhd-dev)
@@ -100,6 +101,11 @@ IF(ENABLE_MANUAL_OR_DOXYGEN)
     ELSE(ENABLE_DOXYGEN_DOT)
         SET(DOXYGEN_HAVE_DOT "NO")
     ENDIF(ENABLE_DOXYGEN_DOT)
+    IF(ENABLE_DOXYGEN_SHORTNAMES)
+        SET(DOXYGEN_SHORT_NAMES "YES")
+    ELSE(ENABLE_DOXYGEN_SHORTNAMES)
+        SET(DOXYGEN_SHORT_NAMES "NO")
+    ENDIF(ENABLE_DOXYGEN_SHORTNAMES)
 
     CONFIGURE_FILE(
         ${CMAKE_CURRENT_SOURCE_DIR}/Doxyfile.in
diff --git a/host/docs/Doxyfile.in b/host/docs/Doxyfile.in
index 8aa6299..d230352 100644
--- a/host/docs/Doxyfile.in
+++ b/host/docs/Doxyfile.in
@@ -147,7 +147,7 @@ STRIP_FROM_INC_PATH    = @CMAKE_SOURCE_DIR@/include
 # (but less readable) file names. This can be useful if your file system
 # doesn't support long names like on DOS, Mac, or CD-ROM.
 
-SHORT_NAMES            = NO
+SHORT_NAMES            = @DOXYGEN_SHORT_NAMES@
 
 # If the JAVADOC_AUTOBRIEF tag is set to YES then Doxygen
 # will interpret the first line (until the first dot) of a JavaDoc-style
-- 
2.1.4

